### PR TITLE
GUI Rewrite

### PR DESCRIPTION
**Pull request by SuperTails**
The rewrite is not perfect yet but seems to be stable enough and contains a lot of very useful features that are very nice to have in the lab. It's also more extensible, so it'd be nice to get this merged soon so new features can be built on top of it.

### CODE REVIEW

File: gui_rewrite.py
Comment (SuperTails) [inline gui_rewrite.py:1]: delete this >:(

File: requirements.txt
Comment (SuperTails) [inline requirements.txt:1]: add toml to this >:(

File: src/hardware.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+           self.commit()
+
+    def commit(self):
+        self.stage.move_to(self.target_position)
+        self.stage_position = copy.copy(self.target_position)
+
+class Lithographer:
+    red_focus: ProcessedImage
+    uv_focus: ProcessedImage
+    pattern: ProcessedImage
```
Comment (SuperTails) [inline src/hardware.py]: probably move these?

**Review by jkentw — APPROVED** [review]
Everything here looks fantastic

**Pull request merged**